File: src/main/java/kanti/sl/arguments/TypeChecker.java
package kanti.sl.arguments;

import org.jetbrains.annotations.NotNull;

public final class TypeChecker {

	@NotNull
	public static Type getType(@NotNull Object value) {
		Class<?> type = value.getClass();
		if (type.equals(Byte.class)) {
			return Type.ConvertToInt;
		} else if (type.equals(Short.class)) {
			return Type.ConvertToInt;
		} else if (type.equals(Integer.class)) {
			return Type.Support;
		} else if (type.equals(Long.class)) {
			return Type.ConvertToInt;
		} else if (type.equals(Float.class)) {
			return Type.ConvertToDouble;
		} else if (type.equals(Double.class)) {
			return Type.Support;
		} else if (type.equals(Boolean.class)) {
			return Type.Support;
		} else if (type.equals(String.class)) {
			return Type.Support;
		} else {
			return Type.Unsupported;
		}
	}

	@NotNull
	public static Object convert(@NotNull Object value, @NotNull Type type) {
		String valueString = value.toString();
		switch (type) {
			case Support:
				return value;
			case ConvertToInt:
				return Integer.parseInt(valueString);
			case ConvertToDouble:
				return Double.parseDouble(valueString);
			case Unsupported:
				Class<?> klazz = value.getClass();
				throw new IllegalStateException("Unsupported type=" + klazz.getSimpleName());
		}
		Class<?> klazz = value.getClass();
		throw new IllegalStateException("Unsupported type=" + klazz.getSimpleName());
	}

	public static boolean check(@NotNull Object value) {
		Class<?> type = value.getClass();
		if (type.equals(Integer.class)) {
			return true;
		} else if (type.equals(Double.class)) {
			return true;
		} else if (type.equals(Boolean.class)) {
			return true;
		} else return type.equals(String.class);
	}

	public enum Type {
		Support,
		ConvertToInt,
		ConvertToDouble,
		Unsupported
	}

}

File: src/main/kotlin/kanti/sl/arguments/TypeChecker.kt
package kanti.sl.arguments

object TypeChecker {

	@JvmStatic
	fun getType(any: Any): Type {
		return when (any) {
			is Byte -> Type.ConvertToInt
			is Short -> Type.ConvertToInt
			is Int -> Type.Support
			is Long -> Type.ConvertToInt
			is Float -> Type.ConvertToDouble
			is Double -> Type.Support
			is Boolean -> Type.Support
			is String -> Type.Support
			else -> Type.Unsupported
		}
	}

	@JvmStatic
	fun check(any: Any): Boolean {
		return when (any) {
			is Int -> true
			is Double -> true
			is Boolean -> true
			is String -> true
			else -> false
		}
	}

	@JvmStatic
	fun convert(value: Any, type: Type): Any {
		return when (type) {
			Type.Support -> {
				value
			}
			Type.ConvertToInt -> {
				value.toString().toInt()
			}
			Type.ConvertToDouble -> {
				value.toString().toDouble()
			}
			Type.Unsupported -> {
				throw IllegalArgumentException("Unsupported type: ${value.javaClass.simpleName}")
			}
		}
	}

	enum class Type {
		Support,
		ConvertToInt,
		ConvertToDouble,
		Unsupported
	}

}